### PR TITLE
Review of Button component

### PR DESCRIPTION
**Pull request by clukhei**
add unit test

fix autogenerate of controls

add button with icon documentation

allow extended props to be laoded into storybook argstable

hide as & bsPrefix prop

create common hidden props helper obj for storybook

### CODE REVIEW

File: .storybook/manager.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ addons.setConfig({
   selectedPanel: undefined,
   initialActive: 'sidebar',
   sidebar: {
-    showRoots: false,
+    showRoots: true,
```
Comment (haphizi) [inline .storybook/manager.js:16]: what's the functionality for this?

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -73,7 +73,7 @@ const defaultProps = {
   disabled: false,
 };
 
-const Button: BsPrefixRefForwardingComponent<'button', ButtonProps> =
+export const Button: BsPrefixRefForwardingComponent<'button', ButtonProps> =
```
Comment (haphizi) [inline src/components/Button/Button.tsx:76]: this one is the one you mention to generate the argstable properly correct?

File: .storybook/manager.js (hunk 1 above)
Comment (clukhei) [reply .storybook/manager.js:16]: changed this but forgot to change it back. Its for showing the folder root directory in left side bar

File: src/components/Button/Button.tsx (hunk 2 above)
Comment (clukhei) [reply src/components/Button/Button.tsx:76]: yes

**Review by rav3 — APPROVED** [review]

**Pull request merged**